stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/mediapreview/MediaGalleryPreviewViewModelFactory.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.viewmodel.mediapreview
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.setup.state.ClientState
+
+/**
+ * Holds the dependencies required for the Media Preview Screen.
+ * Currently builds the [MediaGalleryPreviewViewModel] using those dependencies.
+ *
+ * @param chatClient An instance of the low level client
+ * used for basic chat API functionality.
+ * @param clientState Holds information about the current SDK state.
+ * @param messageId The ID of the message we are fetching in order
+ * to display the attachments.
+ */
+public class MediaGalleryPreviewViewModelFactory(
+    private val chatClient: ChatClient,
+    private val clientState: ClientState = chatClient.clientState,
```

Comment:
Do we really need this property or can we access it below?

Reply:
We can access it through `ChatClient` but I'm just following @leandroBorgesFerreira's advice on making things more testable.

Since we do a lot of our instantiation statically, without doing this any VM tests we write could end up with far more mocking than if we do it this way.